temporalio/converter.py

```diff
@@ -0,0 +1,362 @@
+        type_hint: Optional[Type] = None,
+    ) -> Tuple[Any, bool]:
+        """Decode a single payload to a Python value if able.
+
+        Args:
+            payload: Payload to convert to Python value.
+            type_hints: Type that is expected if any. This may not have a type
+                if there are no annotations on the target.
+
+        Return:
+            A tuple with the first value as the Python value and the second as
+            whether it could be converted or not. If the payload can be
+            converted, it is the first value of the tuple and the second value
+            is True. If the payload cannot be converted, the first value is
+            undefined and the second value is False.
+        """
+        return (None, False)
+
+
+class CompositeDataConverter(DataConverter):
+    """Composite data converter that delegates to a list of payload converters.
+
+    Encoding/decoding are attempted on each payload converter successively until
+    it succeeds.
+
+    Attributes:
+        converters: List of payload converters to delegate to, in order.
+    """
+
+    converters: list[PayloadConverter]
+
+    def __init__(self, *converters: PayloadConverter) -> None:
+        """Initializes the data converter.
+
+        Args:
+            converters: Payload converters to delegate to, in order.
+        """
+        self.converters = list(converters)
+
+    async def encode(self, values: list[Any]) -> list[temporalio.api.common.v1.Payload]:
+        """Encode values trying each converter.
+
+        See base class. Always returns the same number of payloads as values.
+        """
+        payloads = []
+        for index, value in enumerate(values):
+            # We intentionally attempt these serially just in case a stateful
+            # converter may rely on the previous values
+            payload = None
+            for converter in self.converters:
+                payload = await converter.encode(value)
+                if payload is not None:
+                    break
+            if payload is None:
+                raise RuntimeError(
+                    f"Value at index {index} of type {type(value)} could not be converted"
+                )
+            payloads.append(payload)
+        return payloads
+
+    async def decode(
+        self,
+        payloads: list[temporalio.api.common.v1.Payload],
+        type_hints: Optional[list[Type]],
+    ) -> list[Any]:
+        """Decode values trying each converter.
+
+        See base class. Always returns the same number of values as payloads.
+        """
+        values = []
+        for index, payload in enumerate(payloads):
+            type_hint = None
+            if type_hints is not None:
+                type_hint = type_hints[index]
+            # We intentionally attempt these serially just in case a stateful
+            # converter may rely on the previous values
+            ok = False
+            for converter in self.converters:
```

Comment:
This isn't how it works in the other SDKs, see TS as reference: https://github.com/temporalio/sdk-typescript/blob/0bc5ccf412018817894356866e6a46cf878f10f8/packages/common/src/converter/data-converter.ts#L101

Reply:
Ok, I will change to make it where a converter can only convert one encoding. I figured that was less flexible, but I suppose harmless.